haigha/connection.py

```diff
@@ -404,7 +404,10 @@ def read_frames(self):
         # Send a heartbeat (if needed)
         self._channels[0].send_heartbeat()
 
-        data = self._transport.read(self._heartbeat)
+        # Wait for 2 heartbeat intervals before giving up. See AMQP 4.2.7:
+        # "If a peer detects no incoming traffic (i.e. received octets) for two heartbeat intervals or longer,
+        # it should close the connection"
+        data = self._transport.read(self._heartbeat*2)
```

Comment:
Scratch that - it does so only if `Connection._heartbeat` was defined! That's a bug. Should I fix it as well?